app/src/verifier.h

```diff
@@ -66,76 +58,322 @@ namespace scitt::verifier
-            resolution.did_doc, assertion_method_id);
+          throw cose::COSEDecodeError(
+            "Missing io.cncf.notary.signingTime in protected header");
         }
-        catch (const did::DIDAssertionMethodError& e)
+        if (notary_authentic_signing_time.has_value())
         {
-          throw VerificationError(e.what());
+          throw cose::COSEDecodeError(
+            "io.cncf.notary.authenticSigningTime not allowed in protected "
+            "header when io.cncf.notary.signingScheme is `notary.x509`");
+        }
+      }
+      else if (notary_signing_scheme.value() == "notary.x509.signingAuthority")
+      {
+        // notary_authentic_signing_time is critical and required iff
+        // notary.x509.signingAuthority
+        if (notary_signing_time.has_value())
+        {
+          throw cose::COSEDecodeError(
+            "Notary io.cncf.notary.signingTime not allowed in protected header "
+            "when io.cncf.notary.signingScheme is "
+            "`notary.x509.signingAuthority`");
+        }
+        if (!notary_authentic_signing_time.has_value())
+        {
+          throw cose::COSEDecodeError(
+            "Missing io.cncf.notary.authenticSigningTime in protected header");
+        }
+        if (!phdr.is_critical("io.cncf.notary.authenticSigningTime"))
+        {
+          throw cose::COSEDecodeError(
+            "Missing io.cncf.notary.authenticSigningTime in crit parameters");
         }
+      }
+      else
+      {
+        throw cose::COSEDecodeError(
+          "Notary io.cncf.notary.signingScheme must be `notary.x509` or "
+          "`notary.x509.signingAuthority`");
+      }
 
-        // Convert the JWK into something we can actually use.
-        key = get_jwk_public_key(jwk);
+      if (notary_expiry.has_value())
+      {
+        // notary_expiry is critial but not required.
+        if (!phdr.is_critical("io.cncf.notary.expiry"))
+        {
+          throw cose::COSEDecodeError(
+            "Missing io.cncf.notary.expiry in crit parameters");
+        }
       }
+    }
+
+    PublicKey process_notary_profile(
+      cose::UnprotectedHeader& uhdr,
+      cose::ProtectedHeader& phdr,
+      kv::Tx& tx,
+      const Configuration& configuration)
+    {
+      // Validate protected header
+      validate_notary_protected_header(phdr, configuration);
+
+      // Validate unprotected header
+      // Unprotected header must contain x5chain else the notary claim cannot be
+      // verified.
+      if (!uhdr.x5chain.has_value())
+      {
+        throw VerificationError(
+          "Notary claim's unprotected header is missing an x5chain (label 33) "
+          "parameter.");
+      }
+
+      PublicKey key;
+      // Verify the chain of certs against the x509 root store.
+      auto roots = x509_root_store(tx);
+      auto cert = verify_chain(roots, uhdr.x5chain.value());
+      key = PublicKey(cert, std::nullopt);
+      return key;
```

Comment:
The assignment to a variable defined 3 lines up is a bit confusing and unnecessary IMO.
```
return PublicKey(cert, std::nullopt);
```